ops/storage-ddls/cassandra-setup.cql

```diff
@@ -79,87 +76,126 @@ CREATE TYPE features (
 );
 
 CREATE TABLE computedtiles (
+    tileid text,
     tilex int,
     tiley int,
     tilez int,
-    tileid text,
     periodstartdate timestamp,
     periodenddate timestamp,
     periodtype text,
     period text,
-    pipeline text,
-    sourceid text,
-    topic text,
-    topiclangcode text,
+    pipelinekey text,
+    externalsourceid text,
+    topicconjunctionid uuid,
     insertion_time timestamp,
     heatmap text,
     placeids frozen<set<text>>,
     computedfeatures frozen<features>,
-    PRIMARY KEY ((tilex, tiley, tilez), periodstartdate, periodenddate, pipeline, sourceid, topic)
+    PRIMARY KEY ((period, periodtype, pipelinekey, topicconjunctionid), externalsourceid, tilez, tilex, tiley, periodstartdate, periodenddate)
 );
 
-CREATE INDEX ON computedtiles (tileid);
-CREATE INDEX ON computedtiles (period);
-CREATE INDEX ON computedtiles (computedfeatures);
-CREATE INDEX ON computedtiles (periodtype);
+create table topicconjunctions(
+    topic text,
+    topicconjunctionid uuid,
+    conjunctivekeywordcombination map<text, int>,//map value is representative of the aggregated mention count displayed in the dashboard
```

Comment:
I wonder if the mention-count should be in a separate table so that we can partition it by time range. This will enable two use-cases:

1) We'll be able to make the watchlist-term-counts component (see screenshot below) respond to the time-range selected in the dashboard/timeline so that the co-occurrence counts displayed in the component correspond to the term co-occurrences in the current time-period instead of "all time" which is the data stored in this table.

<img src="https://user-images.githubusercontent.com/1086421/28139227-55f4a170-6708-11e7-9c11-b85d56a11925.png" width=200 />
 
2) We'll be able to use the partitioned table to satisfy the data requirements for the time-series query. The time-series component needs the counts of terms bucketed by time-period which is what this table plus a time-partition would give us.

```js
/**
 * @param {{site: string, fromDate: string, toDate: string, zoomLevel: number, limit: number, layertype: string, sourceFilter: string[], mainEdge: string}} args
 * @returns {Promise.<{labels: Array<{name: string, mentions: number}>, graphData: Array<{date: string, edges: string[], mentions: number[]}>}>}
 */
function timeSeries(args, res)
```